src/Fixer/ConfigurationDefinitionFixerInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer;
+
+use PhpCsFixer\OptionsResolver;
+
+/**
+ * @deprecated Will be merged into ConfigurableFixerInterface in 3.0
+ */
+interface ConfigurationDefinitionFixerInterface extends ConfigurableFixerInterface
```

Comment:
I know, the point was to add a new method to `ConfigurableFixerInterface` but this is a BC break, hence the new interface deprecated because it will be merged into the current one in 3.0. I removed the `@deprecated` tag.

Current state is:
* `ConfigurableFixerInterface` not deprecated;
* `ConfigurationDefinitionFixerInterface` extends `ConfigurableFixerInterface` with a new method, not deprecated.

So now I can deprecate `ConfigurableFixerInterface`, making `ConfigurationDefinitionFixerInterface` the interface to use in 3.0. But I think `ConfigurableFixerInterface` is a better name, I would like to keep it in 3.0 and remove `ConfigurationDefinitionFixerInterface` instead. So what about having:

* `ConfigurableFixerInterface` (not ?) deprecated, will have a new method in 3.0;
* `ConfigurationDefinitionFixerInterface` extends `ConfigurableFixerInterface`, not deprecated, will be deprecated in 3.0 and removed in 4.0.

Is such process desirable? worth it? too complicated? Does it make sense to deprecate an interface while it is not to be removed later?